Create Kyma Metrics Collector (#583)
kyma-project/control-plane: components/kyma-metrics-collector/README.md

> ### Kyma Metrics Collector

Reviewer: ```suggestion
# Kyma Metrics Collector
```
Adjusting because of the Kyma README [template](https://github.com/kyma-project/community/blob/main/guidelines/repository-guidelines/repository-template/README.md).

> #### Description

Reviewer: ```suggestion
## Overview
```

Kyma Metrics Collector scrapes all Kyma clusters and uses Shoot information to generate metrics as event streams. The generated event streams are POST-ed to an events collecting system.

> #### Usage

Reviewer: ```suggestion
## Usage
```

> - `Kyma Metrics Collector` comes with the following command line argument flags:

Reviewer: ```suggestion
### Flags

Kyma Metrics Collector comes with the following command line argument flags:
```

    | Flag | Description | Default Value   |
    | ----- | ------------ | --------------- |

>     | `gardener-secret-path` | The path to the secret which contains kubeconfig of the Gardener MPS cluster. | `/gardener/kubeconfig` |

Reviewer: ```suggestion
    | `gardener-secret-path` | The path to the secret which contains the kubeconfig of the Gardener MPS cluster. | `/gardener/kubeconfig` |
```

>     | `gardener-namespace` | The namespace in gardener cluster where information on Kyma clusters are. | `garden-kyma-dev`    |

Reviewer: ```suggestion
    | `gardener-namespace` | The namespace in the Gardener cluster where information on Kyma clusters is. | `garden-kyma-dev`    |
```

>     | `scrape-interval` | The wait duration of the interval between 2 executions of metrics generation. | `3m`         |

Reviewer: ```suggestion
    | `scrape-interval` | The time interval to wait between 2 executions of metrics generation. | `3m`         |
```

    | `worker-pool-size` | The number of workers in the pool. | `5` |

>     | `log-level` | The log-level of the application. E.g. fatal, error, info, debug etc. | `info` |

Reviewer: ```suggestion
    | `log-level` | The log-level of the Application. For example, `fatal`, `error`, `info`, `debug`. | `info` |
```

>     | `listen-addr` | The application starts server in this port to cater to the metrics and health endpoints. | `8080` |

Reviewer: ```suggestion
    | `listen-addr` | The Application starts the server in this port to cater to the metrics and health endpoints. | `8080` |
```

>     | `debug-port` | The custom port to debug when needed. `0` will disable debugging server. | `0` |

Reviewer: ```suggestion
    | `debug-port` | The custom port to debug when needed. `0` will disable the debugging server. | `0` |
```

> - `Kyma Metrics Collector` comes with the following environment variables:

Reviewer: ```suggestion
### Environment variables

Kyma Metrics Collector comes with the following environment variables:
```

     
     | Variable | Description | Default Value   |
     | ----- | ------------ | ------------- |

>      | `PUBLIC_CLOUD_SPECS` | The specification contains the CPU, Network and Disk information for all machine types from a public cloud provider.  | `-` |

Reviewer: ```suggestion
     | `PUBLIC_CLOUD_SPECS` | This specification contains the CPU, Network and Disk information for all machine types from a public cloud provider.  | `-` |
```

     | `KEB_URL` | The KEB URL where Kyma Metrics Collector fetches runtime information. | `-` |

>      | `KEB_TIMEOUT` | The timeout governs the connections from Kyma Metrics Collector to KEB | `30s` |

Reviewer: ```suggestion
     | `KEB_TIMEOUT` | This timeout governs the connections from Kyma Metrics Collector to KEB | `30s` |
```

     | `KEB_RETRY_COUNT` | The number of retries Kyma Metrics Collector will do when connecting to KEB fails. | 5 |

>      | `KEB_POLL_WAIT_DURATION` | The wait duration for Kyma Metrics Collector between each execution of polling KEB for runtime information. | `10m` |

Reviewer: ```suggestion
     | `KEB_POLL_WAIT_DURATION` | The time interval for Kyma Metrics Collector to wait between each execution of polling KEB for runtime information. | `10m` |
```

>      | `EDP_URL` | The EDP base URL where Kyma Metrics Collector will ingest event-stream to. | `-` |

Reviewer: ```suggestion
     | `EDP_URL` | The EDP base URL where Kyma Metrics Collector will ingest the event-stream to. | `-` |
```

     | `EDP_TOKEN` | The token used to connect to EDP. | `-` |

>      | `EDP_NAMESPACE` | The namespace in EDP where Kyma Metrics Collector will ingest event-stream to.| `kyma-dev` |

Reviewer: ```suggestion
     | `EDP_NAMESPACE` | The namespace in EDP where Kyma Metrics Collector will ingest the event-stream to.| `kyma-dev` |
```

>      | `EDP_DATASTREAM_NAME` | The datastream in EDP where Kyma Metrics Collector will ingest event-stream to. | `consumption-metrics` |

Reviewer: ```suggestion
     | `EDP_DATASTREAM_NAME` | The datastream in EDP where Kyma Metrics Collector will ingest the event-stream to. | `consumption-metrics` |
```

     | `EDP_DATASTREAM_VERSION` | The datastream version which Kyma Metrics Collector will use. | `1` |
     | `EDP_DATASTREAM_ENV` | The datastream environment which Kyma Metrics Collector will use.  | `dev` |
     | `EDP_TIMEOUT` | The timeout for Kyma Metrics Collector connections to EDP. | `30s` |
     | `EDP_RETRY` | The number of retries for Kyma Metrics Collector connections to EDP. | `3` |

> #### Development

Reviewer: ```suggestion
## Development
```

> - Run a deployment in currently configured k8s cluster

Reviewer: ```suggestion
- Run a deployment in a currently configured k8s cluster:
```

> 
> ```
> ### It needs a token from a secret kcp-kyma-metrics-collector 
> ko apply -f dev/  
> ```

Reviewer: ````suggestion
>**NOTE:** In order to do this, you need a token from a secret `kcp-kyma-metrics-collector`.
```
ko apply -f dev/  
```
````

> - Resolve all dependencies

Reviewer: ```suggestion
- Resolve all dependencies:
```

```
make gomod-vendor
```

> - Run tests

Reviewer: ```suggestion
- Run tests:
```

```
make tests
```

> - Run tests and publish a test coverage report

Reviewer: ```suggestion
- Run tests and publish a test coverage report:
```

```
make publish-test-results
```

> #### Troubleshooting

Reviewer: ```suggestion
### Troubleshooting
```

> - Check logs

Reviewer: ```suggestion
- Check logs:
```

```
kubectl logs -l app=kyma-metrics-collector -n kcp-system -c kyma-metrics-collector -f
```

> #### Data collection

Reviewer: ```suggestion
### Data collection
```
